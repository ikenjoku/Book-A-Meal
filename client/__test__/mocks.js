
export const meals = [
  {
    id: 0,
    name: 'Nigerian Jollof',
    description: 'Max Jollof rice, plantain, grilled turkey, bellefull Coke',
    price: 1100,
    imageurl: 'https://africa-publicmages/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 1,
    name: 'Rice and Beans',
    description: 'Rice, Beans, Plantain, Panla Sauce, Max Coke',
    price: 1500,
    imageurl: 'https://africa-g/images/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 2,
    name: 'Egusi Semo',
    description: 'Semolina, Egusi Soup, Meat, Extra',
    price: 1200,
    imageurl: 'https://africa-public.food.jcts/80/80418_1465475724_ma.jpg',
  },
  {
    id: 3,
    name: 'Green Calulu',
    description: 'Chicken salad with extra cabbage',
    price: 1250,
    imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 4,
    name: 'Onion Jappatti',
    description: 'Ingera palate with raw meat sauce',
    price: 2200,
    imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
];

export const testMeal = {
  id: 3234,
  name: 'Strawberry flakes',
  description: 'Ingera palate with raw meat sauce',
  price: 2200,
  imageurl: 'https://dynamic/prodg/images/produc475724_ma.jpg',
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const mockStoreData = {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [
      {
        id: 3,
        name: 'Green Calulu',
        description: 'Chicken salad with extra cabbage',
        price: 1250,
        imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      {
        id: 4,
        name: 'Onion Jappatti',
        description: 'Ingera palate with raw meat sauce',
        price: 2200,
        imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    error: null,
    isLoading: false,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders: false,
  },
};

export const createdMenu = {
  message: 'Menu for 2018-06-15 was successfully created',
  newMenu: {
    id: 24,
    date: '2018-06-15',
    updatedAt: '2018-07-02T17:37:46.023Z',
    createdAt: '2018-07-02T17:37:46.023Z',
  },
};

export const menus = [{
  id: 23,
  date: '2018-07-02',
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
},
{
  id: 24,
  date: new Date().toISOString().substr(0, 10),
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
}];

export const orders = [
  {
    id: 2,
    date: '2018-05-16',
    amount: 1500,
    userId: 1,
    mealId: 2,
    status: 'cancelled',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 1,
      firstname: 'Ezinne',
      lastname: 'Njoku',
    },
    Meal: {
      id: 2,
      name: 'Rice and Beans',
      price: 1500,
    },
  },
  {
    id: 3,
    date: '2018-05-16',
    amount: 1200,
    userId: 2,
    mealId: 3,
    status: 'pending',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 2,
      firstname: 'Kelechi',
      lastname: 'Njoku',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
  {
    id: 4,
    date: '2018-05-16',
    amount: 1500,
    userId: 3,
    mealId: 3,
    status: 'delivered',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 3,
      firstname: 'Nnamdi',
      lastname: 'James',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
];
