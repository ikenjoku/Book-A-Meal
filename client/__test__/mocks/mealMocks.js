export const getAllMealsMock = [
  {
    name: 'test meal',
    price: 1230,
    description: 'this is a very nice meal',
    imageurl: 'https://sdfsdjhbsn/sdf.asp',
  },
];
