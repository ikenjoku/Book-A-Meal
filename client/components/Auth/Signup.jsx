import React, { Component } from 'react';
import { signupAUser } from "../../actions/authActions/signup";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

class Signup extends Component {
    state = {
        signUpData: {
            firstname: '',
            lastname: '',
            username: '',
            email: '',
            password: '',
            confirmPassword: ''
        },
        error: ''
    }
    onInputChange = (event) => {
      const { signUpData } = this.state;
      signUpData[event.target.name] = event.target.value.trim();
      this.setState(() => ({ signUpData }));
    }

    onSubmit = (event) => {
      event.preventDefault();
      const { signUpData } = this.state;
      if(!signUpData.firstname || !signUpData.lastname || !signUpData.username || !signUpData.email 
        || !signUpData.password || !signUpData.confirmPassword){
            this.setState(() => ({ error: 'Please fill in all the fields' }));
      } else {
      this.props.signupAUser(signUpData);
      }
    }
    render() {
        return (
            this.props.isLoggedIn === true ?  <Redirect to='/menu'/> :
            <main className="signup-bg">
                {this.state.error &&  <p>{this.state.error}</p> }
                <form className="form-wrapper" onSubmit={this.onSubmit}>
                    <h2 className="center">Sign up</h2>
                    <div>
                        <input
                            className="form-control"
                            type="text"
                            name="firstname"
                            placeholder="First Name"
                            value={this.state.signUpData.firstname}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="text"
                            name="lastname"
                            placeholder="Last Name"
                            value={this.state.signUpData.lastname}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="text"
                            name="username"
                            placeholder="Preferred username"
                            value={this.state.signUpData.username}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="email"
                            name="email"
                            placeholder="Email"
                            value={this.state.signUpData.email}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="password"
                            name="password"
                            placeholder="Password"
                            value={this.state.signUpData.password}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="password"
                            name="confirmPassword"
                            placeholder="Confirm password"
                            value={this.state.signUpData.confirmPassword}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            type="submit"
                            className="btn-control loginsubmitBtn"
                            value="Submit"
                        />
                    </div>
                    <div className="right formText">
                        <p>Already have an account? <Link to="/login">Log In</Link> </p>
                    </div>
                </form>
            </main>
        );
    }
}

const mapDispatchToProps = dispatch => 
  bindActionCreators({signupAUser}, dispatch);

export default connect(null, mapDispatchToProps)(Signup);
