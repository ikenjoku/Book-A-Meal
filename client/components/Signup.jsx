import React from 'react';


export default class Signup extends Component{
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <main className="content">
            <form className="formDesign">
                <h2>Sign In</h2>
                <div>
                    <input 
                      className="input-field control"
                      type="text"
                      name="firstname" 
                      placeholder="First Name" 
                    />
                </div>
                <div>
                    <input 
                      className="input-field control" 
                      type="text" 
                      name="lastname" 
                      placeholder="Last Name" 
                    />
                </div>
                <div>
                    <input 
                      className="input-field control" 
                      type="text" 
                      name="username" 
                      placeholder="Preferred username" 
                    />
                </div>
                <div>
                    <input 
                      className="input-field control" 
                      type="email" 
                      name="email" 
                      placeholder="Email" 
                    />
                </div>
                <div>
                    <input 
                      className="input-field control" 
                      type="password" 
                      name="password" 
                      placeholder="Password" 
                    />
                </div>
                <div>
                    <input
                      class="input-field control" 
                      type="password" 
                      name="confirm-password" 
                      placeholder="Confirm password" 
                    />
                </div>
                <div>
                    <input 
                    type="submit" 
                    class="btn btn-nature submitBtn" 
                    value="submit" 
                    />
                </div>
                <div className="right formText">
                    <p>Already have an account? <a href="login.html">Log In</a> </p>
                </div>
            </form>
        </main>
    );
  }
}
