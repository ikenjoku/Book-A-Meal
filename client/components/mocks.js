

export const sampleOrders = {
  message: 'All orders retrieved successfully',
  orders: [
    {
      id: 2,
      date: '2018-05-16',
      amount: 1500,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 3,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 4,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 1,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 5,
      date: '2018-05-23',
      amount: 1500,
      userId: 2,
      mealId: 2,
      status: 'pending',
      createdAt: '2018-05-23T23:55:05.091Z',
      updatedAt: '2018-05-23T23:55:05.091Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 7,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 6,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 8,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
  ],
};

