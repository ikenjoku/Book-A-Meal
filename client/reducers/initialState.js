export default {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [],
    error: null,
    isLoading: true,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders:false,
  },
};
