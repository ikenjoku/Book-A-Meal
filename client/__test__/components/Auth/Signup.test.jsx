import React from 'react';
import { shallow } from 'enzyme';
import { Signup, mapStateToProps } from '../../../components/Auth/Signup.jsx';
import initialState from '../../../reducers/initialState';
import { orders } from '../../mocks';

describe('Signup Component', () => {
  let wrapper, signupAUser, isLoggedIn, authLoading;
  signupAUser = jest.fn();
  isLoggedIn = false
  authLoading = false;

  test('it should render Signup form correctly', () => {
    wrapper = shallow(
      <Signup
        signupAUser={signupAUser}
        isLoggedIn={isLoggedIn}
        authLoading={authLoading}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onInputChange when input fields change value', () => {
    wrapper = shallow(
      <Signup
        signupAUser={signupAUser}
        isLoggedIn={isLoggedIn}
        authLoading={authLoading}
      />);

    const onInputChangeSpy = jest.spyOn(
      wrapper.instance(), 'onInputChange'
    );
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'chubby@gmail.com',
        name: 'email'
      }
    };
    wrapper.instance().onInputChange(event);
    expect(onInputChangeSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle onSubmit call when form is submitted', () => {
    wrapper = shallow(
      <Signup
        signupAUser={signupAUser}
        isLoggedIn={isLoggedIn}
        authLoading={authLoading}
      />);

    const onSubmitSpy = jest.spyOn(
      wrapper.instance(), 'onSubmit'
    );
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.instance().onSubmit(event);
    expect(onSubmitSpy).toHaveBeenCalledTimes(1);
  });

  it('should redirect to menu page if user signs up', () => {
    wrapper = shallow(
      <Signup
        signupAUser={signupAUser}
        isLoggedIn={true}
        authLoading={authLoading}
      />);
      expect(wrapper).toBeDefined();
      expect(wrapper.find('Redirect').length).toBe(1);
      expect(wrapper.find('div').length).toBe(0);
  });

  test('should map state to props', () => {
    const ownProps = { isLoggedIn: false, authLoading: false };
    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });
});
