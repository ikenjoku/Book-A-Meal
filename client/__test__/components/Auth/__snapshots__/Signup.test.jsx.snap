// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component it should render Signup form correctly 1`] = `
<main
  className="signup-bg"
>
  <form
    className="form-wrapper"
    onSubmit={[Function]}
  >
    <h2
      className="center"
    >
      Sign up
    </h2>
    <div>
      <input
        className="form-control"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="form-control"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        placeholder="Preferred username"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div>
      <input
        className="form-control"
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm password"
        type="password"
        value=""
      />
    </div>
    <div>
      <input
        className="btn-control loginsubmitBtn"
        type="submit"
        value="Submit"
      />
    </div>
    <div
      className="right formText"
    >
      <p>
        Already have an account? 
        <Link
          replace={false}
          to="/login"
        >
          Log In
        </Link>
         
      </p>
    </div>
  </form>
</main>
`;

exports[`Signup Component should map state to props 1`] = `
Object {
  "authLoading": true,
  "isLoggedIn": false,
}
`;
